spec/Appendix A -- Persisted Documents.md

```diff
@@ -95,6 +95,50 @@ implementation specific.
 Note: A 32 character hexadecimal _custom document identifier_ is likely to be an
 MD5 hash of the GraphQL document, as traditionally used by Relay.
 
+### Document identifier syntax
+
+DocumentIdentifier ::
+
+- PrefixedDocumentIdentifier
+- CustomDocumentIdentifier
+
+PrefixedDocumentIdentifier ::
+
+- Sha256HexDocumentIdentifier
+- OtherPrefixedDocumentIdentifier
+
+Sha256HexDocumentIdentifier :: `sha256:` LowerCaseHexDigit+
+
+LowerCaseHexDigit :: one of
+
+- `0` `1` `2` `3` `4` `5` `6` `7` `8` `9`
+- `a` `b` `c` `d` `e` `f`
+
+OtherPrefixedDocumentIdentifier :: `x-` UnreservedCharacter+ `:`
+UnreservedCharacter+
+
+CustomDocumentIdentifier :: UnreservedCharacter+
+
+UnreservedCharacter ::
+
+- Letter
+- Digit
+- `-`
+- `.`
+- `_`
+- `~`
```

Comment:
Note: these 'Unreserved Characters' mirror the list within RFC 3986 here:

https://datatracker.ietf.org/doc/html/rfc3986#section-2.3

ALPHA and DIGIT are described further here:

https://datatracker.ietf.org/doc/html/rfc2234#section-6.1

Reply:
This encoding allows for GUIDs encoded as described by RFC 4122:

https://datatracker.ietf.org/doc/html/rfc4122#section-3

And it allows for base64url encoding as described by RFC 4648: (assuming no padding)

https://datatracker.ietf.org/doc/html/rfc4648#section-5

Reply:
Addressed [here](https://github.com/graphql/graphql-over-http/pull/296/files#diff-9be5577e05ae2112d2b8f95584b162d0dec01453bf6c85df58bf5db4f2c9727aR100)